components/scream/src/share/field/field_positivity_check.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef SCREAM_FIELD_POSITIVITY_CHECK_HPP
+#define SCREAM_FIELD_POSITIVITY_CHECK_HPP
+
+#include "share/field/field.hpp"
+#include "ekat/util/ekat_math_utils.hpp"
+
+namespace scream
+{
+
+// This field property check returns true if all data in a given field are
+// positive, and false if not. If constructed with a specific lower bound, it
+// can repair a field that fails the check by setting all nonpositive values
+// to that lower bound. If no specific lower bound is given (i.e. if the
+// default constructor is used), no repairs can be made.
+template<typename ScalarType, typename Device>
+class FieldPositivityCheck: public FieldPropertyCheck<ScalarType, Device> {
```

Comment:
Do you think it's worth to have an impl independent on that of FieldBoundednessCheck? Why not inheriting from that one, and set ub=std::numeric_limits<ScalarType>::max()? It might do additional checks (the upper bound) so there may be a small perf hit. Might not be really noticeable though, and it would keep code repetition lower.

Reply:
Nvm, I now see that this has a sligthly different way of repairing, by setting a value that is strictly larger than 0.

Follow up then: do you think it's worth making a similar repair fix for FieldBoundsCheck? Like, if you passs 3 or 4 scalars, we interpret the 3rd (and 4th, if present) as the "repair" values. If a repair value is provided, they must be strictly contained in the (lb,ub) range; also, if, say, a lb repair value is present, our check is > rather than >=.

Maybe this can be done in a follow up pr though...

Reply:
This is an interesting idea. But I like your suggestion of doing it in a follow-up PR when we might have some discussion of what people expect to be able to do.